pkg/storage/inmemory/article_test.go

```diff
@@ -0,0 +1,338 @@
+			articles, err := storage.GetByKeyword(tt.keyword)
+
+			if tt.expectErr {
+				assert.Error(t, err)
+			} else {
+				assert.NoError(t, err)
+				assert.Equal(t, tt.expected, articles)
+			}
+		})
+	}
+}
+
+func TestArticleInMemory_GetBySource(t *testing.T) {
+	tests := []struct {
+		name      string
+		source    string
+		articles  []model.Article
+		expected  []model.Article
+		expectErr bool
+	}{
+		{
+			name:   "Get articles by source successfully",
+			source: "Source 1",
+			articles: []model.Article{
+				{Id: 1, Title: "Article 1", Description: "Description 1", Link: "http://link1.com", Source: "Source 1", PubDate: time.Date(2023, 6, 15, 0, 0, 0, 0, time.UTC)},
+				{Id: 2, Title: "Another Article", Description: "Description 2", Link: "http://link2.com", Source: "Source 2", PubDate: time.Date(2023, 6, 15, 0, 0, 0, 0, time.UTC)},
+			},
+			expected: []model.Article{
+				{Id: 1, Title: "Article 1", Description: "Description 1", Link: "http://link1.com", Source: "Source 1", PubDate: time.Date(2023, 6, 15, 0, 0, 0, 0, time.UTC)},
+			},
+			expectErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			storage := &ArticleInMemory{Articles: tt.articles}
+			articles, err := storage.GetBySource(tt.source)
+
+			if tt.expectErr {
+				assert.Error(t, err)
+			} else {
+				assert.NoError(t, err)
+				assert.Equal(t, tt.expected, articles)
+			}
+		})
+	}
+}
+
+func TestArticleInMemory_SaveAll(t *testing.T) {
+	tests := []struct {
+		name      string
+		articles  []model.Article
+		expected  []model.Article
+		expectErr bool
+	}{
+		{
+			name: "Save all articles successfully",
+			articles: []model.Article{
+				{Id: 1, Title: "Article 1", Description: "Description 1", Link: "http://link1.com", Source: "Source 1", PubDate: time.Date(2023, 6, 15, 0, 0, 0, 0, time.UTC)},
+				{Id: 2, Title: "Article 2", Description: "Description 2", Link: "http://link2.com", Source: "Source 2", PubDate: time.Date(2023, 6, 15, 0, 0, 0, 0, time.UTC)},
+			},
+			expected: []model.Article{
+				{Id: 1, Title: "Article 1", Description: "Description 1", Link: "http://link1.com", Source: "Source 1", PubDate: time.Date(2023, 6, 15, 0, 0, 0, 0, time.UTC)},
+				{Id: 2, Title: "Article 2", Description: "Description 2", Link: "http://link2.com", Source: "Source 2", PubDate: time.Date(2023, 6, 15, 0, 0, 0, 0, time.UTC)},
+			},
+			expectErr: false,
+		},
+		{
+			name:      "Save empty list of articles",
+			articles:  []model.Article{},
+			expected:  []model.Article{},
+			expectErr: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			storage := New()
+			err := storage.SaveAll(tt.articles)
+
+			if tt.expectErr {
+				assert.Error(t, err)
+			} else {
+				assert.NoError(t, err)
+				assert.Equal(t, tt.expected, storage.Articles)
+			}
+		})
+	}
+}
+
+func TestNew(t *testing.T) {
+	tests := []struct {
+		name string
+		want *ArticleInMemory
+	}{
+		// TODO: Add test cases.
```

Comment:
needed ? if not pls remove.

Reply:
My bad 😅, removed. And I'm not sure, should I keep logic for Deletion of articles, if it is not used now?